File: backend/app/schemas/comment.py
from pydantic import BaseModel, Field
from typing import List, Optional
from .user import UserInfo


class CommentItem(BaseModel):
    """评论响应模型：单条评论内容（含所属游戏 id）"""
    id: int
    content: Optional[str] = Field(default=None, max_length=2000)
    receive_id: int


class CommentItems(BaseModel):
    item: CommentItem
    account: UserInfo


class CommentList(BaseModel):
    total: int
    items: List[CommentItems]


class AddComment(BaseModel):
    """新增评论请求模型：仅 content，接收对象通过路由参数 game_id 传入"""
    content: Optional[str] = Field(default=None, min_length=1, max_length=2000)


class EditComment(BaseModel):
    """编辑评论请求模型：仅允许修改内容"""
    content: Optional[str] = Field(default=None, min_length=1, max_length=2000)

File: frontend/types/comment.ts
// ===== comment 模块类型（与 backend schemas/comment.py 对齐）=====

export interface CommentItem {
  id: number
  content: string | null
  receive_id: number
}

export interface CommentItems {
  item: CommentItem
  account: {
    username: string
    avatar: string | null
  }
}

/** GET /api/v1/comment/{game_id} 列表响应 */
export interface CommentList {
  total: number
  items: CommentItems[]
}

/** POST /api/v1/comment/add 请求体 */
export interface AddComment {
  content: string
}

/** PUT /api/v1/comment/edit 请求体 */
export interface EditComment {
  content: string
}
